compiler-rt/lib/radsan/radsan.cpp

```diff
@@ -7,26 +7,137 @@
 */
 
 #include <radsan/radsan.h>
+
 #include <radsan/radsan_context.h>
+#include <radsan/radsan_flags.h>
 #include <radsan/radsan_interceptors.h>
-#include <unistd.h>
+
+#include "sanitizer_common/sanitizer_atomic.h"
+#include "sanitizer_common/sanitizer_common.h"
+#include "sanitizer_common/sanitizer_flag_parser.h"
+#include "sanitizer_common/sanitizer_flags.h"
+
+using namespace __sanitizer;
+
+namespace radsan {
+static __sanitizer::atomic_uint8_t radsan_inited{};
+static __sanitizer::atomic_uint8_t radsan_init_is_running{};
+static __sanitizer::atomic_uint64_t radsan_report_count{};
+static Flags radsan_flags{};
+
+Flags *flags() { return &radsan_flags; }
+
+void EnsureInitialized() {
+  CHECK(!radsan::IsInitRunning());
+  if (!IsInitialized()) {
+    radsan_init();
+  }
+}
+
+bool IsInitialized() { 
+  return __sanitizer::atomic_load(&radsan_inited, memory_order_acquire) == 1; 
+}
+
+bool IsInitRunning() { 
+  return __sanitizer::atomic_load(&radsan_init_is_running, memory_order_acquire) == 1; 
+}
+
+__sanitizer::u64 GetReportCount() { 
+  return __sanitizer::atomic_load(&radsan_report_count, memory_order_acquire); 
+}
+
+void IncrementReportCount() { 
+  __sanitizer::atomic_fetch_add(&radsan_report_count, 1, memory_order_relaxed); 
+}
+
+
+SANITIZER_INTERFACE_WEAK_DEF(const char *, __radsan_default_options, void) {
+  return "";
+}
+
+
+static void RegisterRadsanFlags(FlagParser *parser, Flags *f) {
+#define RADSAN_FLAG(Type, Name, DefaultValue, Description) \
+  RegisterFlag(parser, #Name, Description, &f->Name);
+#include "radsan_flags.inc"
+#undef RADSAN_FLAG
+}
+
+void Flags::SetDefaults() {
+#define RADSAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;
+#include "radsan_flags.inc"
+#undef RADSAN_FLAG
+}
+
+
+static void initializeFlags() {
+  SetCommonFlagsDefaults();
+  {
+    CommonFlags cf;
+    cf.CopyFrom(*common_flags());
+    cf.stack_trace_format = "DEFAULT";
+    cf.external_symbolizer_path = GetEnv("RADSAN_SYMBOLIZER_PATH");
```

Comment:
Shall we move this into `RADSAN_OPTIONS` as well, for consistency?

Reply:
Actually, we are in good company here! This one is a different env variable for each of the other sanitizers

```
> rg "external_symbolizer_path.*_SYMBOLIZER_PATH"
compiler-rt/lib/ubsan/ubsan_flags.cpp
53:    cf.external_symbolizer_path = GetFlag("UBSAN_SYMBOLIZER_PATH");

compiler-rt/lib/memprof/memprof_flags.cpp
52:    cf.external_symbolizer_path = GetEnv("MEMPROF_SYMBOLIZER_PATH");

compiler-rt/lib/asan/asan_flags.cpp
57:    cf.external_symbolizer_path = GetEnv("ASAN_SYMBOLIZER_PATH");

compiler-rt/lib/radsan/radsan.cpp
83:    cf.external_symbolizer_path = GetEnv("RADSAN_SYMBOLIZER_PATH");

compiler-rt/lib/tsan/rtl/tsan_flags.cpp
60:    cf.external_symbolizer_path = GetEnv("TSAN_SYMBOLIZER_PATH");

compiler-rt/lib/msan/msan.cpp
152:    cf.external_symbolizer_path = GetEnv("MSAN_SYMBOLIZER_PATH");

compiler-rt/lib/lsan/lsan.cpp
57:    cf.external_symbolizer_path = GetEnv("LSAN_SYMBOLIZER_PATH");

compiler-rt/lib/hwasan/hwasan.cpp
78:    cf.external_symbolizer_path = GetEnv("HWASAN_SYMBOLIZER_PATH");
```